Documentation/Evolution/RegexLiteralPitch.md

```diff
@@ -0,0 +1,257 @@
+# Regular Expression Literals
+
+- Authors: Hamish Knight, Michael Ilseman
+
+## Introduction
+
+We propose to introduce a first-class regular expression literal into the language that can take advantage of library support to offer extensible, powerful, and familiar textual pattern matching.
+
+This is a component of a larger string processing picture. We would like to start a focused discussion surrounding our approach to the literal itself, while acknowledging that evaluating the utility of the literal will ultimately depend on the whole picture (e.g. supporting API). To aid this focused discussion, details such as the representation of captures in the type system, semantic details, extensions to lexing/parsing, additional API, etc., are out of scope of this pitch and thread. Feel free to continue discussion of anything related in the [overview thread][overview].
+
+## Motivation
+
+Regular expressions are a ubiquitous, familiar, and concise syntax for matching and extracting text that satisfies a particular pattern. Syntactically, a regex literal in Swift should:
+
+- Be familiar and facilitate knowledge reuse
+- Fit in with Swift's library extensibility story for literals
+
+Further motivation, examples, and discussion can be found in the [overview thread][overview].
+
+## Proposed Solution
+
+We propose the introduction of a regular expression literal that supports [the PCRE syntax][PCRE], in addition to new standard library protocols `ExpressibleByRegexLiteral` and `RegexLiteralProtocol` that allow for the customization of how the regex literal is interpreted (similar to [string interpolation][stringinterpolation]). The compiler will parse the PCRE syntax within a regex literal, and synthesize calls to corresponding builder methods. Types conforming to `ExpressibleByRegexLiteral` will be able to provide a builder type that opts into supporting various regex constructs through the use of normal function declarations and `@available`.
```

Comment:
Could we also support UTS#18? http://www.unicode.org/reports/tr18/

AFAIK PCRE doesn't support all of that, but it'd be good if we could at least parse it.

Reply:
Let's continue talk about TR18 syntax on the thread